src/clients/near.ts

```diff
@@ -564,4 +569,162 @@ export class NearBridgeClient {
+        transfer_id: {
+          origin_chain: args.transfer_id.origin_chain,
+          origin_nonce: args.transfer_id.origin_nonce.toString(),
+        },
+        msg: args.msg,
+        storage_deposit_amount: args.storage_deposit_amount,
+        relayer: args.relayer,
+      },
+    }
+
+    const transferArgs = {
+      receiver_id: this.lockerAddress,
+      amount: args.amount,
+      msg: JSON.stringify(message),
+    }
+
+    // Execute the fast finalize transfer
+    const tx = await this.wallet.functionCall({
+      contractId: args.token_id,
+      methodName: "ft_transfer_call",
+      args: transferArgs,
+      gas: GAS.FAST_FIN_TRANSFER,
+      attachedDeposit: DEPOSIT.INIT_TRANSFER,
+    })
+
+    return tx.transaction.hash
+  }
+
+  /**
+   * Performs a complete fast transfer from EVM to NEAR.
+   * This function orchestrates the entire fast transfer process:
+   * 1. Fetches and parses the InitTransfer event from the EVM transaction
+   * 2. Gets the corresponding NEAR token ID using getBridgedToken
+   * 3. Executes the fast finalize transfer on NEAR
+   *
+   * @param originChain - The EVM chain where the original transfer was initiated
+   * @param evmTxHash - Transaction hash of the InitTransfer on the EVM chain
+   * @param evmClient - EVM bridge client for parsing the transaction
+   * @param storageDepositAmount - Optional storage deposit amount in yoctoNEAR
+   * @returns Promise resolving to the NEAR transaction hash
+   * @throws {Error} If the origin chain is not supported for fast transfers
+   * @throws {Error} If the EVM transaction or event cannot be found/parsed
+   * @throws {Error} If the fast transfer execution fails
+   */
+  async nearFastTransfer(
+    originChain: ChainKind,
+    evmTxHash: string,
+    evmClient: EvmBridgeClient,
+    storageDepositAmount?: string,
+  ): Promise<string> {
+    // Validate supported chains for fast transfer
+    if (!isEvmChain(originChain)) {
+      throw new Error(`Fast transfer is not supported for chain kind: ${ChainKind[originChain]}`)
+    }
+
+    // Step 1: Parse the InitTransfer event from EVM transaction
+    const transferEvent = await evmClient.parseInitTransferEvent(evmTxHash)
+
+    // Step 2: Get the NEAR token ID for the EVM token using getBridgedToken
+    const omniTokenAddress = omniAddress(originChain, transferEvent.tokenAddress)
+    const nearTokenAddress = await getBridgedToken(omniTokenAddress, ChainKind.Near)
+
+    if (!nearTokenAddress) {
+      throw new Error(`No bridged token found on NEAR for ${omniTokenAddress}`)
+    }
+
+    const nearTokenId = nearTokenAddress.split(":")[1] // Extract account ID from near:account.near
+
+    // Step 3: Construct the transfer ID
+    const transferId: TransferId = {
+      origin_chain: originChain, // Use numeric enum value
+      origin_nonce: transferEvent.originNonce,
+    }
+
+    // Step 4: Execute the fast finalize transfer
+    const fastTransferArgs: FastFinTransferArgs = {
```

Comment:
`amount` and `fee` need to be converted according to the decimal difference. They need to be submitted with Near decimals instead of the origin values. This is also an issue in rust SDK if that was the reference